node/network/statement-distribution/src/lib.rs

```diff
@@ -1928,25 +1970,50 @@ struct MetricsInner {
 	active_leaves_update: prometheus::Histogram,
 	share: prometheus::Histogram,
 	network_bridge_update_v1: prometheus::Histogram,
+	statements_unexpected: prometheus::CounterVec<prometheus::U64>,
+}
+
+// Used to distinguish different unexpected statements in the metrics
+enum UnexpectedStatementLabel {
+	Seconded,
+	Valid,
+	Large,
+}
+
+impl From<UnexpectedStatementLabel> for &'static str {
+	fn from(label: UnexpectedStatementLabel) -> Self {
+		match label {
+			UnexpectedStatementLabel::Seconded => "seconded",
+			UnexpectedStatementLabel::Valid => "valid",
+			UnexpectedStatementLabel::Large => "large",
+		}
+	}
 }
 
 /// Statement Distribution metrics.
 #[derive(Default, Clone)]
 pub struct Metrics(Option<MetricsInner>);
 
 impl Metrics {
+	/// Update statements distributed counter
```

Comment:
nit: With content breaking the 2k loc, it might make sense to move this into a separate file `metrics.rs`

Reply:
Done.